### PR TITLE
Feature/cvsb 5862

### PR DESCRIPTION
**Pull request by oalexandru**

### CODE REVIEW

File: tests/unit/TestResultsService.unitTest.js [hunk 1]
```diff
@@ -628,25 +628,75 @@ describe('setExpiryDateAndCertificateNumber', () => {
...
-          expect((response.testTypes[0].testExpiryDate).split('T')[0]).to.equal(expectedExpiryDate.toISOString().split('T')[0])
-          expect(response.testTypes[0].certificateNumber).to.equal(response.testTypes[0].testNumber)
-          expect(response.testTypes[1].testExpiryDate).to.equal(undefined)
-          expect(response.testTypes[1].certificateNumber).to.equal(undefined)
-          expect(response.testTypes[2].testExpiryDate).to.equal(undefined)
-          expect(response.testTypes[2].certificateNumber).to.equal(undefined)
-        })
+    context('for psv vehicle type', () => {
+      it('should set the expiryDate and the certificateNumber for "Annual With Certificate" testTypes with testResult "pass", "fail" or "prs"', () => {
+        testResultsDAOMock.testResultsResponseMock = Array.of(testResultsMockDB[0])
```
Comment (arupdvsa) [inline tests/unit/TestResultsService.unitTest.js]: use beforeEach for test initialisation , apparently are failing if existing initialisation pattern is subjected to parallel tests
So below goes into beforeEach(...) just above the it(..)

testResultsDAOMock.testResultsResponseMock = Array.of(testResultsMockDB[0])
        const testResultsService = new TestResultsService(testResultsDAOMock)
       testResultsDAOMock.numberOfrecords = ''		         let mockData = testResultsMockDB[0]
       testResultsDAOMock.numberOfScannedRecords = ''		         testResultsDAOMock.numberOfrecords = ''
       mockData.testTypes[2].testResult = ''		         testResultsDAOMock.numberOfScannedRecords = ''
       return testResultsService.setExpiryDateAndCertificateNumber(mockData)		         mockData.testTypes[2].testResult = ''

File: tests/unit/TestResultsService.unitTest.js [hunk 2]
```diff
@@ -684,7 +734,7 @@ describe('insertTestResultsList', () => {
 
       return testResultsService.insertTestResultsList(testResultsDAOMock.testResultsResponseMock)
         .then(data => {
```
Comment (arupdvsa) [inline tests/unit/TestResultsService.unitTest.js]: specify data with type

File: tests/unit/TestResultsService.unitTest.js [hunk 3]
```diff
@@ -726,7 +776,7 @@ describe('deleteTestResultsList', () => {
 
       return testResultsService.deleteTestResultsList(testResultsDAOMock.testResultsResponseMock)
         .then(data => {
```
Comment (arupdvsa) [inline tests/unit/TestResultsService.unitTest.js]: specify data with type

File: tests/unit/TestResultsService.unitTest.js [hunk 4]
```diff
@@ -628,25 +628,75 @@ describe('setExpiryDateAndCertificateNumber', () => {
...
+          return testResultsService.setExpiryDateAndCertificateNumber(mockData)
+            .then(response => {
+              const expectedExpiryDate = dateFns.addYears(dateFns.lastDayOfMonth(new Date()), 1)
+              expect((response.testTypes[0].testExpiryDate).split('T')[0]).to.equal(expectedExpiryDate.toISOString().split('T')[0])
+            })
+        })
+      })
+      context('when there is a certificate issued for this vehicle that is expiring more than two months in the future', () => {
+        it('should set the expiry date to last day of current month + 1 year', () => {
+          testResultsDAOMock.testResultsResponseMock = Array.of(testResultsMockDB[15])
```
Comment (arupdvsa) [inline tests/unit/TestResultsService.unitTest.js]: initialise mocks in beforeEach to allow for parallel tests

File: tests/unit/TestResultsService.unitTest.js (hunk 2 above)
Comment (oalexandru) [reply tests/unit/TestResultsService.unitTest.js]: can't specify type in javascript

File: tests/unit/TestResultsService.unitTest.js (hunk 3 above)
Comment (oalexandru) [reply tests/unit/TestResultsService.unitTest.js]: can't specify type in javascript

File: tests/unit/TestResultsService.unitTest.js (hunk 4 above)
Comment (oalexandru) [reply tests/unit/TestResultsService.unitTest.js]: done

File: tests/unit/TestResultsService.unitTest.js (hunk 1 above)
Comment (oalexandru) [reply tests/unit/TestResultsService.unitTest.js]: done

**Pull request merged**